### PR TITLE
feat:sort out the status and API of the layout

### PR DESCRIPTION
**Pull request by lichunn**
English | [简体中文](https://github.com/opentiny/tiny-engine/blob/develop/.github/PULL_REQUEST_TEMPLATE/PULL_REQUEST_TEMPLATE.zh-CN.md)

# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-engine/blob/develop/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Built its own designer, fully self-validated

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

### What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

## Summary by CodeRabbit

- **New Features**
  - Introduced a new `Lock` component to enhance toolbar functionality.
  - Expanded layout configuration with new options for visibility and toolbar arrangement.

- **Improvements**
  - Enhanced state management in toolbar components for better user interaction.
  - Improved organization of toolbar items for clearer management.
  - Adjusted styling for toolbar elements, including increased spacing for better layout.

- **Bug Fixes**
  - Updated popover trigger behavior for more intuitive user interaction.

### CODE REVIEW

File: packages/layout/src/DesignToolbars.vue [hunk 1]
```diff
@@ -30,36 +46,71 @@ export default {
     toolbars: {
       type: Array,
       default: () => []
+    },
+    layoutRegistry: {
+      type: Object,
+      default: () => {}
```
Comment (hexqi) [inline packages/layout/src/DesignToolbars.vue:48]: 少了括号， () => ({})

File: packages/layout/src/DesignToolbars.vue [hunk 2]
```diff
@@ -30,36 +46,71 @@ export default {
...
+        }
+      })
+
+      return res
+    }
+
+    const getNewBar = (registryData, stataData) =>
+      Array.from(new Set([...registryData, ...stataData])).map((id) => getMergeRegistry(REGISTRY_NAME, id))
+
     props.toolbars.forEach((item) => {
```
Comment (hexqi) [inline packages/layout/src/DesignToolbars.vue:83]: 缩进有问题

File: packages/layout/src/DesignToolbars.vue [hunk 3]
```diff
@@ -30,36 +46,71 @@ export default {
...
+      newArr.forEach((id) => {
+        if (arr.indexOf(id) === -1) {
+          res.push(id)
+        }
+      })
+
+      return res
+    }
+
+    const getNewBar = (registryData, stataData) =>
```
Comment (hexqi) [inline packages/layout/src/DesignToolbars.vue]: 是不拼错了，stataData --> stateData

File: packages/layout/src/DesignToolbars.vue [hunk 4]
```diff
@@ -30,36 +46,71 @@ export default {
...
       } else if (item.align === 'center') {
-        centerBar.push(item)
+        state.centerBar.push(item.id)
       } else {
-        leftBar.push(item)
+        state.leftBar.push(item.id)
       }
     })
+
+    state.leftBar = getNewBar(registryToolbars.left, state.leftBar)
```
Comment (hexqi) [inline packages/layout/src/DesignToolbars.vue]: getNewBar这个逻辑感觉不太对哦，如果代码里面写的是左侧, 配置的是在右侧，那是否会在左右侧都显示呢？

File: packages/layout/src/ToolbarCollapse.vue [hunk 5]
```diff
@@ -6,11 +6,11 @@
           <svg-icon name="ellipsis"></svg-icon>
         </span>
       </template>
-      <div class="empty-bar" v-for="item in collapseBar" :key="item.id">
-        <div v-if="item.splitLine" class="empty-line"></div>
-        <div class="toolbar-list-button">
-          <component :is="item.entry"></component>
+      <div class="empty-bar" v-for="(item, idx) in collapseBar" :key="idx">
+        <div class="toolbar-list-button" v-for="comp in item" :key="comp">
+          <component :is="getMergeRegistry(registry, comp)?.entry"></component>
```
Comment (hexqi) [inline packages/layout/src/ToolbarCollapse.vue]: 可以改用getMergeMeta，传入id参数即可

File: packages/layout/src/DesignToolbars.vue (hunk 4 above)
Comment (lichunn) [reply packages/layout/src/DesignToolbars.vue]: 是左右侧都显示，这里没有去限制用户同个工具只能显示在某一侧；
如果用户在代码里和在配置都写在同一侧，才会去去重

File: designer-demo/registry.js [hunk 6]
```diff
@@ -49,14 +50,31 @@ import engineConfig from './engine.config'
...
+    ...Layout,
+    options: {
+      ...Layout.options,
+      isShowLine: true,
+      isShowCollapse: true,
+      toolbars: {
+        left: ['engine.toolbars.breadcrumb', 'engine.toolbars.lock', 'engine.toolbars.logo'],
+        center: ['engine.toolbars.media'],
+        right: [
+          ['engine.toolbars.clean'],
```
Comment (hexqi) [inline designer-demo/registry.js]: 插件顺序跟设计稿好像有点差异

File: packages/toolbars/breadcrumb/src/Main.vue [hunk 7]
```diff
@@ -94,16 +91,21 @@ export default {
   align-items: center;
   width: auto;
   height: 100%;
+  margin-right: 3px;
+  cursor: pointer;
```
Comment (hexqi) [inline packages/toolbars/breadcrumb/src/Main.vue]: 面包屑那里的指针有点问题，页面名称那里还是光标

File: designer-demo/registry.js (hunk 6 above)
Comment (lichunn) [reply designer-demo/registry.js]: options的配置和toolbars的配置是可以合并的，假设toolbars的右侧配置比options的右侧多了一个组件，会自动放到右侧分组第一组，这里Redoundo只写在toolbars是为了测试

File: packages/layout/src/DesignToolbars.vue (hunk 4 above)
Comment (hexqi) [reply packages/layout/src/DesignToolbars.vue]: 但是合并的逻辑看起来不太对，你试下我上面列的这种场景，应该会出现同一个插件在左右两侧都渲染的情况

**Review by hexqi — APPROVED** [review]

**Pull request merged**